src/librustc_codegen_ssa/debuginfo/type_names.rs

```diff
@@ -190,11 +190,19 @@ pub fn push_debuginfo_type_name<'tcx>(
             // processing
             visited.remove(t);
         },
-        ty::Closure(..) => {
+        ty::Closure(def_id, ..) => {
             output.push_str("closure");
+            let disambiguator = tcx.def_key(def_id).disambiguated_data.disambiguator;
+            if disambiguator != 0 {
```

Comment:
I think I would prefer appending the disambiguator unconditionally.

Reply:
Done.